pkg/descheduler/evictions/evictions.go

```diff
@@ -565,9 +565,14 @@ func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, opts EvictOptio
 
 // return (ignore, err)
 func (pe *PodEvictor) evictPod(ctx context.Context, pod *v1.Pod) (bool, error) {
-	deleteOptions := &metav1.DeleteOptions{
-		GracePeriodSeconds: pe.gracePeriodSeconds,
+	deleteOptions := &metav1.DeleteOptions{}
+
+	// Honor the pod's TerminationGracePeriodSeconds firstly.
+	// If the gracePeriodSeconds of pod is not set, then set our value.
+	if pod.Spec.TerminationGracePeriodSeconds == nil {
```

Comment:
this will never be nil, since default value is 30

https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/

>The default terminationGracePeriodSeconds setting is 30 seconds.

Reply:
Yes, correct, this value will never be nil. This ` pod.Spec.TerminationGracePeriodSeconds == nil ` is meaningless.

@googs1025 could you pls tell the reason why adding `gracePeriodSeconds` args?